libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1318,11 +1318,39 @@ bool TypeChecker::visit(Conditional const& _conditional)
 	return false;
 }
 
+void TypeChecker::checkExpressionAssignment(Type const& _type, Expression const& _expression)
+{
+	if (auto const* tupleExpression = dynamic_cast<TupleExpression const*>(&_expression))
+	{
+		if (auto const* tupleType = dynamic_cast<TupleType const*>(&_type))
+		{
+			for (size_t i = 0; i < min(tupleExpression->components().size(), tupleType->components().size()); i++)
+				if (tupleType->components()[i] && tupleExpression->components()[i])
```

Comment:
Please assert that `!tupleType->components()[i] == !tupleExpression->components()[i]`

Reply:
Here we can't assert, though - e.g. ``(x,) = (1,);`` is a non-fatal error, so the type may still be null.